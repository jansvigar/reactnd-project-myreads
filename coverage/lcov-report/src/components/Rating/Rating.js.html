<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Rating/Rating.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/components/Rating</a> Rating.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/10</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import FontAwesome from 'react-fontawesome';
import PropTypes from 'prop-types';
import 'font-awesome/css/font-awesome.min.css';
&nbsp;
const Rating = props =&gt; {
    const ratingStyle = {
      color: '#FFB300'
    }
&nbsp;
    const starFull = &lt;FontAwesome name="star" size="lg" style={ ratingStyle } /&gt;;
    const starHalf = &lt;FontAwesome name="star-half-o" size="lg" style={ ratingStyle }/&gt;;
    const starEmpty = &lt;FontAwesome name="star-o" size="lg" style={ ratingStyle } /&gt;;
&nbsp;
    const isFractionalNumber = Math.ceil(props.number) !== props.number;
&nbsp;
    return (
      &lt;div&gt;
        {
          Array.apply(null, Array(5))
               .map((item, index, array) =&gt; {
                 return (index + 1) &lt;= props.number
                    ? <span class="branch-0 cbranch-no" title="branch not covered" >&lt;span key={ index }&gt;{ starFull }&lt;/span&gt;</span>
                    : isFractionalNumber &amp;&amp; (<span class="branch-1 cbranch-no" title="branch not covered" >Math.ceil(props.number) === (index + 1))</span>
                    ? <span class="branch-0 cbranch-no" title="branch not covered" >&lt;span key={ index }&gt;{ starHalf }&lt;/span&gt;</span>
                    : &lt;span key={ index }&gt;{ starEmpty }&lt;/span&gt;
               })
&nbsp;
        }
      &lt;/div&gt;
    );
};
&nbsp;
Rating.propTypes = {
  number: PropTypes.number
}
&nbsp;
Rating.defaultProps = {
  number: 0
}
&nbsp;
export default Rating;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Aug 11 2017 00:33:58 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
