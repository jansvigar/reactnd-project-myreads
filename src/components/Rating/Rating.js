import React from 'react';
import FontAwesome from 'react-fontawesome';
import PropTypes from 'prop-types';
import 'font-awesome/css/font-awesome.min.css';

const Rating = props => {
    const ratingStyle = {
      color: '#FFB300'
    }

    const starFull = <FontAwesome name="star" size="lg" style={ ratingStyle } />;
    const starHalf = <FontAwesome name="star-half-o" size="lg" style={ ratingStyle }/>;
    const starEmpty = <FontAwesome name="star-o" size="lg" style={ ratingStyle } />;

    const isFractionalNumber = Math.ceil(props.number) !== props.number;

    return (
      <div>
        {
          Array.apply(null, Array(5))
               .map((item, index, array) => {
                 return (index + 1) <= props.number
                    ? <span key={ index }>{ starFull }</span>
                    : isFractionalNumber && (Math.ceil(props.number) === (index + 1))
                    ? <span key={ index }>{ starHalf }</span>
                    : <span key={ index }>{ starEmpty }</span>
               })

        }
      </div>
    );
};

Rating.propTypes = {
  number: PropTypes.number
};

Rating.defaultProps = {
  number: 0
};

export default Rating;
