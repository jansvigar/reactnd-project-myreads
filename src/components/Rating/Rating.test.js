import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Rating from './Rating';

describe('<Rating />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Rating />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
