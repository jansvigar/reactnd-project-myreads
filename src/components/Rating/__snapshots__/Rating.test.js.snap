// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating /> should render correctly 1`] = `
<div>
  <span>
    <FontAwesome
      name="star-o"
      size="lg"
      style={
        Object {
          "color": "#FFB300",
        }
      }
    />
  </span>
  <span>
    <FontAwesome
      name="star-o"
      size="lg"
      style={
        Object {
          "color": "#FFB300",
        }
      }
    />
  </span>
  <span>
    <FontAwesome
      name="star-o"
      size="lg"
      style={
        Object {
          "color": "#FFB300",
        }
      }
    />
  </span>
  <span>
    <FontAwesome
      name="star-o"
      size="lg"
      style={
        Object {
          "color": "#FFB300",
        }
      }
    />
  </span>
  <span>
    <FontAwesome
      name="star-o"
      size="lg"
      style={
        Object {
          "color": "#FFB300",
        }
      }
    />
  </span>
</div>
`;
